2024-01-28 02:32:54,891: root: INFO: MLTrader : Executing the initialize lifecycle method
2024-01-28 02:33:05,865: root: INFO: Current backtesting datetime 2020-01-02 08:30:00-05:00
2024-01-28 02:33:05,866: root: INFO: MLTrader : Executing the before_market_opens lifecycle method
2024-01-28 02:33:05,866: root: INFO: Current backtesting datetime 2020-01-02 09:30:00-05:00
2024-01-28 02:33:05,866: root: INFO: MLTrader : Executing the before_starting_trading lifecycle method
2024-01-28 02:33:05,868: root: INFO: [34mMLTrader : Executing the on_trading_iteration lifecycle method at 2024-01-28 02:33:05[0m
2024-01-28 02:33:05,870: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 02:33:05,873: root: WARNING: quote is not implemented for YahooData, but USD was passed as the quote
2024-01-28 02:33:05,874: root: ERROR: type object 'datetime.datetime' has no attribute 'datetime'
2024-01-28 02:33:05,875: root: ERROR: Traceback (most recent call last):
  File "C:\Users\Ishan Roy Barman\AppData\Roaming\Python\Python311\site-packages\lumibot\strategies\strategy_executor.py", line 829, in run
    self._run_trading_session()
  File "C:\Users\Ishan Roy Barman\AppData\Roaming\Python\Python311\site-packages\lumibot\strategies\strategy_executor.py", line 780, in _run_trading_session
    self._on_trading_iteration()
  File "C:\Users\Ishan Roy Barman\AppData\Roaming\Python\Python311\site-packages\lumibot\strategies\strategy_executor.py", line 274, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ishan Roy Barman\AppData\Roaming\Python\Python311\site-packages\lumibot\strategies\strategy_executor.py", line 298, in func_output
    result = func_input(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ishan Roy Barman\AppData\Roaming\Python\Python311\site-packages\lumibot\strategies\strategy_executor.py", line 87, in func_output
    return func_input(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ishan Roy Barman\AppData\Roaming\Python\Python311\site-packages\lumibot\strategies\strategy_executor.py", line 378, in _on_trading_iteration
    on_trading_iteration()
  File "C:\Users\Ishan Roy Barman\AppData\Roaming\Python\Python311\site-packages\lumibot\tools\decorators.py", line 62, in func_output
    frame, result = call_function_get_frame(func_input, *args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ishan Roy Barman\AppData\Roaming\Python\Python311\site-packages\lumibot\tools\decorators.py", line 30, in call_function_get_frame
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Ishan Roy Barman\Desktop\algo\main.py", line 58, in on_trading_iteration
    probability, sentiment = self.get_sentiment()
                             ^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Ishan Roy Barman\Desktop\algo\main.py", line 40, in get_sentiment
    start_date = max(three_days_prior, datetime.datetime(2020, 1, 1))  # Replace with the actual start date
                                       ^^^^^^^^^^^^^^^^^
AttributeError: type object 'datetime.datetime' has no attribute 'datetime'

2024-01-28 02:33:05,876: root: INFO: MLTrader : Executing the on_bot_crash event method
2024-01-28 02:33:05,891: root: INFO: Backtesting finished
2024-01-28 02:33:05,892: root: WARNING: Cannot plot returns because the strategy returns are missing
2024-01-28 02:33:05,948: root: WARNING: Cannot create a tearsheet because the strategy returns are missing
2024-01-28 02:33:05,948: backtest_stats: INFO: Backtest took 0:00:11.057803 for a speed of 0.000
